v-next/hardhat/src/internal/builtin-plugins/solidity/build-system/compilation-job.ts

```diff
@@ -71,23 +88,41 @@ export class CompilationJobImplementation implements CompilationJob {
     return this.#resolvedFiles;
   }
 
-  #buildSolcInput(): CompilerInput {
+  #buildSolcInputAndCoverageMetadata(): [CompilerInput, CoverageMetadata] {
     const solcInputWithoutSources = this.#getSolcInputWithoutSources();
 
     const sources: { [sourceName: string]: { content: string } } = {};
+    let coverageMetadata: CoverageMetadata = {};
 
     const resolvedFiles = this.#getResolvedFiles();
 
     for (const file of resolvedFiles) {
-      sources[file.sourceName] = {
-        content: file.content.text,
-      };
+      if (this.#coverage && file.type === ResolvedFileType.PROJECT_FILE) {
+        // TODO: Call EDR to instrument the content of the file and obtain the metadata
```

Comment:
Please note that this TODO will be filled in once the instrumentation feature is available in EDR. This will be added in a future PR.